src/pd/client.rs

```diff
@@ -17,33 +17,93 @@ use std::time::Duration;
 use std::sync::Mutex;
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::thread;
+use std::collections::HashSet;
 use util::codec::rpc;
 use util::make_std_tcp_conn;
 
 use rand::{self, Rng};
 
-use kvproto::pdpb::{Request, Response};
+use kvproto::pdpb::{self, Request, Response};
 use kvproto::msgpb::{Message, MessageType};
 
-use super::{Result, PdClient};
+use super::{Result, PdClient, protocol};
 use super::metrics::*;
 
 const MAX_PD_SEND_RETRY_COUNT: usize = 100;
 const SOCKET_READ_TIMEOUT: u64 = 3;
 const SOCKET_WRITE_TIMEOUT: u64 = 3;
 
 const PD_RPC_PREFIX: &'static str = "/pd/rpc";
+const MSG_ID_VALIDATE: u64 = 0;
```

Comment:
so you should use another cluster var for cluster id. 
Btw, the ID here is not `INVALID`.